src/blueteam/Robot.java

```diff
@@ -37,6 +40,67 @@ void run() throws GameActionException {
 	abstract void step() throws GameActionException;
 
 	/**
+	 * advertise to others that this robot is alive
+	 */
+	private void newRobotBorn() {
+		alive = true;
+		updateRobotCount(1);
+	}
+
+	/**
+	 * advertise imminent death to all friends, so they could mourn their buddy
+	 */
+	private void checkHealth() {
+		if (alive && rc.getHealth() < rc.getType().getStartingHealth() * TeamConstants.MINIMUM_HEALTH_PERCENTAGE) {
```

Comment:
Recounting the number of units each turn would be safer. If a scout (10 HP total) is killed by a soldier (2HP shot), it will never report it's dying. (Here, it is just a matter of < vs <=, but a unit can be hit by more than one shot at once.) 
On the other hand, it is a good idea to hire reserve soldiers in advance. Perhaps, it would be better to increase the MINIMUM_HEALTH_PERCENTAGE to e. g. 0.41?